tests/rptest/tests/pandaproxy_test.py

```diff
@@ -72,6 +72,24 @@ def _create_topics(self,
                           replication_factor=replicas))
         return names
 
+    def _wait_for_topic(self, name):
+        kc = KafkaCat(self.redpanda)
+        has_leaders = False
+        while not has_leaders:
+            topics = kc.metadata()["topics"]
+            maybe_leaders = True
+            for t in topics:
+                if t["topic"] == name:
+                    for p in t["partitions"]:
+                        if p["leader"] == -1:
+                            maybe_leaders = False
+            has_leaders = maybe_leaders
+        # TODO:
+        #  Despite the above test, Pandaproxy can still get back no leaders
+        #  Query Pandaproxy metadata to see when leaders have settled
+        #  The retry logic for produce should have sufficient time for this
+        #  additional settle time.
```

Comment:
when you working on this TODO you probably also want to make use of the back-off logic in`from ducktape.utils.util import wait_until`. there are several example uses in the tree.

there is also a helper `self.redpanda.partitions("topic")` that you can use to simplify your loop.